Sources/SpeziFHIRHealthKit/FHIRResource+HealthKit.swift

```diff
@@ -0,0 +1,71 @@
+//
+// This source file is part of the Stanford Spezi open source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+@preconcurrency import HealthKit
+import HealthKitOnFHIR
+import ModelsDSTU2
+import ModelsR4
+import SpeziFHIR
+
+
+extension FHIRResource {
+    /// Creates a new ``FHIRResource`` instance using an `HKSample`.
+    /// - Parameters:
+    ///   - sample: The sample that should be transformed in a ``FHIRResource``.
+    ///   - hkHealthStore: Optional `HKHealthStore` used to query additional context such as symptoms and voltage measurements for electrocardiograms. Set to nil to avoid this behavior.
+    /// - Returns: Created ``FHIRResource`` instance.
+     public static func initialize(
```

Comment:
All the things happening here seem like a good case for using the Strategy pattern. Each case representing a strategy.